### PR TITLE
#163043221 - code refactoring based on feedback

### PR DESCRIPTION
**Pull request by openwell**
#### What does this PR do?
- This PR refactors my middleware.js and meetUpModel.js code base
#### Description of Task to be completed?
- 

#### How should this be manually tested?
- Pull this branch [ch-refactoring-code-based-feedback-163043221](https://github.com/openwell/sendIT/tree/ch-refactoring-code-based-feedback-163043221) off this repository
- On your command line, run npm install to install all dependencies for this app
- On your command line, run node npm start to initialize the app
- App would run on port 3000

#### Any background context you want to provide?
- This app is a node app and would require nodejs and npm installation on your local machine
#### What are the relevant pivotal tracker stories?
[163043221](https://www.pivotaltracker.com/story/show/163043221)
#### Screenshots (if appropriate)
None
#### Questions:
None

### CODE REVIEW

File: src/helpers/middleware.js [hunk 1]
```diff
@@ -1,57 +1,109 @@
...
-      const errors = validationResult(req);
-      if (!errors.isEmpty()) {
-        return res.status(400).json({
-          status: 400,
-          errors: errors.array(),
-        });
-      }
-      return next();
-    },
+    check('id')
```
Comment (ludralph) [inline src/helpers/middleware.js]: Good work on raising this PR however your code should be DRY. Consider refactoring. Too much repetition here.

Comment (openwell) [reply src/helpers/middleware.js]: Alright

File: src/helpers/middleware.js [hunk 2]
```diff
@@ -1,57 +1,78 @@
...
+      errors: errors.array(),
+    });
+  }
+  return next();
+};
+
+function validationHandler(arg) {
+  return check(arg)
+    .escape()
+    .isLength({ min: 1, max: 4 })
```
Comment (ludralph) [inline src/helpers/middleware.js]: Consider passing min and max as arguments

File: src/helpers/middleware.js [hunk 3]
```diff
@@ -1,57 +1,78 @@
...
+function validationHandler1(arg) {
+  return check(arg)
+    .escape()
+    .isLength({ min: 5, max: 10 })
+    .withMessage('must be minimum of 5-10 letters')
+    .isString()
+    .withMessage('must be an string');
+}
+
+function validationHandler2(arg) {
```
Comment (ludralph) [inline src/helpers/middleware.js]: If validation handler is done right we will eliminate this repeated function

File: src/helpers/middleware.js [hunk 4]
```diff
@@ -1,57 +1,59 @@
...
+function validationHandler(arg, min, max) {
+  return check(arg)
+    .escape()
+    .isLength({ min, max })
+    .withMessage(`must be minimum of ${min} -${max} letters`)
+    .isInt()
+    .withMessage('must be an integer');
+}
+
+function validationHandler1(arg, min, max) {
```
Comment (ludralph) [inline src/helpers/middleware.js]: minor change is needed rename validationHandler1 to validationHandlerForStringInput and validationHandler to validationHandlerForIntegerInput

**Review by ludralph — APPROVED** [review]
LGTM

**Pull request merged**